### PR TITLE
Issues/1944

### LINKED ISSUE #1944: Cache direct queries and objects
## Issue Overview

Direct queries made with `$wpdb', Give_Donor and Give_Payment do not cache automatically. We can cache them to improve performance.
For example, Donor related queries currently are uncached.
( I will update this after some more research )

Similar #2121

## Todos
- [ ] Tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This will resolve
#1944 
#2464

## How Has This Been Tested?

Manually by performing following operations:
1. Form update
2. Donor create/update/delete
3. Payment update
4. .... [Addon testing] 

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/settings/class-settings-advanced.php [hunk 1]
```diff
@@ -52,6 +52,28 @@ public function get_settings() {
...
+							'desc'    => __( 'If caching enabled then plugin start caching custom post type related queries and reduce load from your site.', 'give' ),
+							'id'      => 'cache',
+							'type'    => 'radio_inline',
+							'default' => 'enabled',
+							'options' => array(
+								'enabled'  => __( 'Enabled', 'give' ),
+								'disabled' => __( 'Disabled', 'give' ),
+							),
+						),
+						array(
```
Comment (mehul0810) [inline includes/admin/settings/class-settings-advanced.php]: @ravinderk This settings ( i.e. Remove Data on Uninstall) is duplicated in this PR. This happens when i add this PR to Give `release 2.0`

**Check Screenshot for reference:**
image

**Comment by mehul0810** [comment]
@ravinderk I think we should add option to show the status of `Caching` under System Info. This will help us in debugging when any customer report any issue after Give `2.0` is LIVE. Let me know your thoughts.

**Comment by ravinderk** [comment]
@mehul0810 Good Idea

**Comment by ravinderk** [comment]
@mehul0810 Added.
screen shot 2017-12-06 at 6 24 37 pm

**Comment by emgk** [comment]
@ravinderk I am getting PHP notice when open add donation form page in the backend.

`Notice: Undefined index: key in /app/public/wp-content/plugins/Give/includes/payments/backward-compatibility.php on line 583` 

https://github.com/ravinderk/Give/blob/issues/1944/includes/payments/backward-compatibility.php#L583

**Pull request merged**